README.md

```diff
@@ -26,15 +26,35 @@ Open http://localhost:6274/?transport=streamable-http&serverUrl=http://mcp.127-0
 
 ## Example OAuth setup
 
-After running the Quick start above,
-create the EnvoyFilter with a hardcoded oauth-protected-resource response,
-and an AuthPolicy that validates tokens on the /mcp endpoint.
+After running the Quick start above, configure OAuth by setting environment variables for the mcp-broker and applying an AuthPolicy that validates tokens on the /mcp endpoint.
+
+Configure the mcp-broker with OAuth environment variables:
 
 ```bash
-kubectl apply -f ./config/istio/gateway/oauth-envoyfilter.yaml
+# Configure OAuth discovery endpoint via environment variables
+export OAUTH_RESOURCE_NAME="MCP Server"
+export OAUTH_RESOURCE="http://mcp.127-0-0-1.sslip.io:8888/mcp"  
+export OAUTH_AUTHORIZATION_SERVERS="http://keycloak.127-0-0-1.sslip.io:8888/realms/mcp"
+export OAUTH_BEARER_METHODS_SUPPORTED="header"
+export OAUTH_SCOPES_SUPPORTED="basic"
+
+# Restart the broker to pick up the new configuration
+kubectl set env deployment/mcp-broker-router \
+  OAUTH_RESOURCE_NAME="$OAUTH_RESOURCE_NAME" \
+  OAUTH_RESOURCE="$OAUTH_RESOURCE" \
+  OAUTH_AUTHORIZATION_SERVERS="$OAUTH_AUTHORIZATION_SERVERS" \
+  OAUTH_BEARER_METHODS_SUPPORTED="$OAUTH_BEARER_METHODS_SUPPORTED" \
+  OAUTH_SCOPES_SUPPORTED="$OAUTH_SCOPES_SUPPORTED" \
+  -n mcp-system
+
+# Apply AuthPolicy for token validation
 kubectl apply -f ./config/mcp-system/authpolicy.yaml
 ```
 
+The mcp-broker now serves OAuth discovery information at `/.well-known/oauth-protected-resource`.
+
+### Keycloak Setup
+
 Set up a new 'mcp' realm in keycloak:
```

Comment:
wonder if we could/should automate this in setup for demo-ability.

Reply:
I'll look into options for having a script to do this